python/ray/air/config.py

```diff
@@ -15,12 +16,13 @@
     from ray.tune.stopper import Stopper
     from ray.tune.execution.placement_groups import PlacementGroupFactory
 
-ScalingConfig = Dict[str, Any]
+
+SampleRange = Union[Domain, Dict[str, List], List]
```

Comment:
@xwjiang2010 this concept was originally introduced in your PR - do we actually need all these types? Why not just `Domain`?

Reply:
when copying @xwjiang2010's code my understanding was `Dict[str, List]` allows `{"grid_search": list_of_values}`, and figured that `List` was also meant for grid search values

That said, do we even need the `SampleRange` type here at all? since we resolve tune vars only after converting to dict, the `ScalingConfig` dataclass will only hold scalar values, right?

Reply:
Oh huh I never even knew that was possible 😆 Is this in our public documentation?! (I didn't see it in the [user guide](https://docs.ray.io/en/latest/tune/tutorials/tune-search-spaces.html) or [API docs](https://docs.ray.io/en/latest/tune/api_docs/search_space.html)).

Few alternatives that make sense to me:
1. Make this a first class concept in Tune (it seems odd to me to have this defined in AIR but not in Tune).
2. Get rid of this functionality and restrict usage to Domain.

@xwjiang2010 @krfricke thoughts?

Reply:
The code for `grid_search` is very simple:

```
def grid_search(values: Iterable) -> Dict[str, List]:
    """Convenience method for specifying grid search over a value.

    Arguments:
        values: An iterable whose parameters will be gridded.
    """
    return {"grid_search": values}
```

That said, it is an internal representation and users should always use `tune.grid_search`. We may change this representation at some point.

The `List` can be used for any kind of nested value, so it's recursive `List[SampleRange]` (which is invalid to define). That said it will yield a list, so if none of our properties are lists, we can remove it here.

For the question at hand, users should always just use `tune.grid_search`, but we need to allow `dict` here so IDEs play nicely with it.